text/metrics/0156-counter-gauge-model.md

```diff
@@ -0,0 +1,356 @@
+instrumentation library, resource attributes, and metric attributes.
+
+_Metric data points_ are the items in a stream, each has a point
+kind. For the purpose of this text, the point kind is Sum or Gauge.
+Sum points have two options: Temporality and Monotonicity.
+
+_Metric timeseries_ is the output of aggregating a stream of data
+points for a specific set of resource and attribute dimensions.
+
+## Meaning and interpretation of Counter and UpDownCounter events
+
+Counter and UpDownCounter instruments produce Sum metric data
+points that are taken to have meaning in a metric stream, independent
+of the aggregation temporality, as follows:
+
+- Sum points are quantities that define a rate of change with respect to time
+- Rate-of-change over time combined with a reset time may be used to derive a current total.
+
+The rate interpretation is preferred for monotonic Sum points, and the
+the current total interpretation is preferred for non-monotonic Sum
+points.  Both interpretations are meaningful and useful for both kinds
+of Sum point.
+
+Sum points imply a linear scale of measurement.  A Sum value that is
+twice the amount of another actually means twice as much of the
+variable was counted.  Linear interpolation is considered to preserve
+meaning.
+
+## Meaning and interpretation of Gauge events
+
+Gauge instruments produce Gauge metric data points are taken to
+have meaning in a metric stream as follows:
+
+- Gauge point values are individual measurements captured at an instant in time
+- Gauge points record the last known value in a series of individual measurements.
+
+Note that these two statements imply different interpretation for
+synchronous and asynchronous measurements.  When recording Gauge
+values through a synchronous API, the interpretation is "last known
+value", and when recording Gauge values through an asynchronous API
+the interpretation is "current value".
+
+The distinction between last known value (synchronous) and current
+value (asynchronous) is considered not significant in the data model.
+Contrasting with Sum points, less can be assumed about the
+measurements.  No implied linear scale of measurement, therefore:
+
+- Rate-of-change may not be well defined
+- Ratios are not necessarily meaningful
+- Linear interpolation is not necessarily supported.
+
+## Attributes are used for interpretation
+
+Metric attributes enable new ways to interpret a stream of metric
+data.  Metric attributes add information without changing the value of
+a metric event.  Addition and removal of metric attributes can be
+accomplished safely by applying transformations that preserve meaning.
+
+Addition of attributes on a metric event can create new timeseries, by
+producting a of greater number of distinct attribute sets.  However,
+the meaning in the original events is preserved in the complete set of
+timeseries.
+
+Removing attributes from metric streams without changing meaning
```

Comment:
I'd suggest moving this down below the example, and adding a sentence afterward to draw attention to the problem, e.g. 

"...which means applying the natural aggregation function to merge metric streams. But how do we know which is the "natural" aggregation function? For Counters the answer is always SUM (for: reason), however for Gauges it might be SUM or MEAN depending on the semantics of the values represented by the particular metric. (for: reason). Therein lies the problem."